lib/game.rb

```diff
@@ -57,10 +57,60 @@ def valid_move(index)
     end
   end
 
+  def horizontal_win
+    top_row_win = (@board.new_board[0] != '1') && three_all_same_token(@board.new_board[0, 3])
+    middle_row_win = (@board.new_board[3] != '4') && three_all_same_token(@board.new_board[3, 3])
+    bottom_row_win = (@board.new_board[6] != '7') && three_all_same_token(@board.new_board[6, 3])
+    if top_row_win || middle_row_win || bottom_row_win
```

Comment:
The expression `top_row_win || middle_row_win || bottom_row_win` itself evaluates to true/false, so you can skip the if/else expression:

```ruby
def horizontal_win
  # ...
  top_row_win || middle_row_win || bottom_row_win
end
```

Reply:
this change has been updated to all my win logic